adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1843,94 +977,95 @@ void deserialize(final String serializedBlob) throws AuthenticationException {
         }
 
         if (mBrokerProxy.canSwitchToBroker()) {
-            throw new UsageAuthenticationException(ADALError.FAIL_TO_IMPORT,"Failed to import the serialized blob because broker is enabled.");
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_IMPORT, "Failed to import the serialized blob "
+                    + "because broker is enabled.");
         }
 
         final TokenCacheItem tokenCacheItem = SSOStateSerializer.deserialize(serializedBlob);
         final String cacheKey = CacheKey.createCacheKey(tokenCacheItem);
         this.getCache().setItem(cacheKey, tokenCacheItem);  
     }
 
-    class DefaultConnectionService implements IConnectionService {
+    void setIsAuthorityValidated(final boolean isAuthorityValidated) {
+        mIsAuthorityValidated = isAuthorityValidated;
+    }
+
+    boolean getIsAuthorityValidated() {
+        return mIsAuthorityValidated;
+    }
 
-        private Context mConnectionContext;
+    AuthenticationRequestState getWaitingRequest(final int requestId) throws AuthenticationException {
+        Logger.v(TAG, "Get waiting request: " + requestId);
+        AuthenticationRequestState request = null;
 
-        DefaultConnectionService(Context ctx) {
-            mConnectionContext = ctx;
+        mReadLock.lock();
+        try {
+            request = mDelegateMap.get(requestId);
+        } finally {
+            mReadLock.unlock();
         }
```

Comment:
I would prefer to keep the Reentrantlock instead of using Synchronized. 
1) synchronization does a full lock on the object, so reading locks it and writing lock it. 
2) Reentrantlock allows multiple threads to read an object at the same time until a thread wishes to write it. When a thread writes to that object, it is locked from all other threads until the writing is finished. Afterwards, it is released, all the threads are able to read from it again. 
Sure, Reentrantlock performs better under high contention, it may not be the case for our usage. But I don't see the downside of using it for a map.